nixos/doc/manual/release-notes/rl-2103.xml

```diff
@@ -172,6 +188,18 @@
       please note that your non-root pools will now be forcibly imported.
     </para>
    </listitem>
+   <listitem>
+    <para>
+     <option>users.defaultUserShell</option> has been renamed to
+     <link linkend="opt-users.defaults.shell">users.defaults.shell</link>.
+    </para>
```

Comment:
(Unless I'm mistaken, not tested yet)

In addition, the value must now be set to a package.

* * *

The previous implementation used:

```
      type = types.either types.path types.shellPackage;
```

While now it only uses `types.shellPackage`.

The documentation stated:

> This option defines the default shell assigned to user accounts. This can be either a full system path or a shell package.
> 
> This must not be a store path, since the path is used outside the store (in particular in /etc/passwd).

Reply:
This one's intentional. I added some [fairly elaborate backcompat code](https://github.com/NixOS/nixpkgs/pull/101127/commits/66071448860f56e9461d4ce10e8dbe3e38ad293a#diff-e9c7b883f114501b3e493bb2690f3ea51606a925cb91c50da0d7db0ab7992568R49-R63) to take care of the most common cases, and the description explains how to change the path if that isn't enough. The reason why this was even an `either` in the first place was actually due to [further backcompat](https://github.com/NixOS/nixpkgs/commit/2974b6f4c8224ab1097e9386d6b3a0d570f95531) — hopefully 4 years is enough time for it to be reasonable to remove after. Additionally, it's less of a problem restricting the type slightly here, since people will have to update their configs after this change regardless.